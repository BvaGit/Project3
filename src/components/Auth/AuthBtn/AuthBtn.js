import React from "react";

import Button from "../../Button";

const AuthBtn = ({handleAuth, isError}) => {
    return (
        <Button
            type="button"
            textId="SIGN IN"
            disabled={!isError}
            onClick={handleAuth}
        />
    );
}

export default AuthBtn;
