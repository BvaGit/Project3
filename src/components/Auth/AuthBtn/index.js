import {connect} from "react-redux";

import AuthBtn from "./AuthBtn";
import { getIsValidAuthFields } from "../../../store/auth/selectors";
import {authUserRequest} from "../../../store/auth/asyncActions";

const mapStateToProps = state => ({
    isError: getIsValidAuthFields(state),
})

const mapDispatchToProps = dispatch => ({
    handleAuth: () => dispatch(authUserRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AuthBtn);
