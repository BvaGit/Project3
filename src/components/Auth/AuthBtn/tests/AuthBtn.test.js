import React from 'react';
import { shallow, mount } from 'enzyme';
import AuthBtn from '../AuthBtn'
import mapStateToProps from '../index'

describe('AuthBtn', () => {
    it("Should match snapshot", () => {
        const component = shallow(<AuthBtn />)
        expect(component).toMatchSnapshot()
    })
})

describe('mapStateToProps', () =>{
    it('should be defined', function(){
        expect(mapStateToProps).toBeDefined()
    })
})