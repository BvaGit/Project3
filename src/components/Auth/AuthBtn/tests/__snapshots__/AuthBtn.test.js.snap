// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthBtn Should match snapshot 1`] = `ShallowWrapper {}`;
